src/WealthsimpleAPIBase.php

```diff
@@ -185,7 +190,7 @@ private function checkOAuthToken(?callable $persist_session_fct = NULL) : void {
                 $this->session->access_token = $response->access_token;
                 $this->session->refresh_token = $response->refresh_token;
                 if ($persist_session_fct) {
-                    $persist_session_fct($this->session);
+                    $persist_session_fct($this->session, self::$username);
```

Comment:
Mixing `$this` and `self` is not a good idea. Better to add an optional 3rd parameter to the `fromToken()` function & pass it as a new 2nd parameter of `checkOAuthToken()`.

If you just remove from this PR this usage of the static variable (including the 2 next suggestions too), I can add another commit to implement what I mean.

```suggestion
```

Reply:
Ha that's what I originally did, but decided to go this way cause it seemed less disruptive of your design. I'll implement your suggestion cause I do also think it is better.